source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/UpdateHistoricalNumbersSubForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { FieldArray, useFormikContext } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+
+import { LinkButton, RemoveButton } from '@/components/common/buttons';
+import { Input, Select } from '@/components/common/form';
+import { SectionField } from '@/components/common/Section/SectionField';
+import * as API from '@/constants/API';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { exists } from '@/utils';
+
+import { HistoricalNumberForm, UpdatePropertyDetailsFormModel } from './models';
+
+export interface IUpdateHistoricalNumbersSubFormProps {
+  propertyId: number;
+}
+
+export const UpdateHistoricalNumbersSubForm: React.FC<IUpdateHistoricalNumbersSubFormProps> = ({
+  propertyId,
+}) => {
+  const { values, setFieldValue } = useFormikContext<UpdatePropertyDetailsFormModel>();
+  const { getOptionsByType } = useLookupCodeHelpers();
+
+  // (sort alpha; exceptions: 1. Other at end, and 2. PS second in list)
+  // The order is set via displayOrder in the DB
+  const historicalNumberTypes = getOptionsByType(API.HISTORICAL_NUMBER_TYPES);
+
+  return (
+    <FieldArray
+      name="historicalNumbers"
+      render={arrayHelpers => (
+        <>
+          {values.historicalNumbers?.map((hn, index) => (
+            <React.Fragment key={`property-historical-${index}`}>
+              <Row className="py-3" data-testid={`historicalNumberRow[${index}]`}>
+                <Col xs="auto" xl="5">
+                  <Input field={`historicalNumbers.${index}.historicalNumber`} />
+                </Col>
+                <Col xs="auto" xl="5" className="pl-0">
+                  <Select
+                    data-testid="select-historical-type"
```

Comment:
this is the format we generally use for data-testids. (lowercase, hyphenated). would be nice to update line 35 to this format.